### ISSUE TITLE
Automatic template generation

### ISSUE THREAD

**Issue by Infel1** [HDoujinDownloader/HDoujinDownloader#398]
Well. I found a way so that you don't have to create a template for each of the websites, of course the ones that are “the same as each other” will not require a template for each one. What you will do is that the websites that are already incorporated into HDoujin, the program will generate a template for these sites in case they change domains so you won't have to reincorporate them into the program. Unless you find a definitive way for Hdoujin to detect website structures and chapters along with titles, I consider this the best way to not have to deal with the same websites over and over again. Only a few use the “Domain Management”. then it is not valid; That option will be removed in future updates. We already had a similar discussion in one of my questions. Still, I want to give more feedback so that the program can improve. What we should prioritize is to automate the process of adding websites or at least make it more efficient and semi-automatic.

**Comment by HDoujinDownloader** [comment]
I know we talked about support for "generic" modules in #317. I mentioned that they're now properly supported in #388, but hadn't uploaded any module updates to take advantage of it yet.

I just updated the module for Ikigai Mangas to detect the template they use: 31df992

It basically does as you said: Looks for some indicators to decide what template the site is using, and handles it accordingly. As long as they keep using the same template, all new domains will be detected and you won't need to update anything manually.

I'm not able to test this right now, though, because none of the images are loading for me on this particular site, even in a web browser. I'm not sure if it's IP-based or if the site just broken. Let me know if you're experiencing similar issues.

**Comment by Infel1** [comment]
https://lectorikigai.efope.com/ This is the new Domain of Ikigai.  If you do not see the images, it will be because the chapters have not been uploaded. Then I will send you several websites so you can add a template to each one and HDoujin will do the rest. Yesterday one of the websites I use changed the domain  I had to use “Domain Management” to reincorporate ti.

**Comment by HDoujinDownloader** [comment]
You're right, the site appears to be working now. We'll continue the discussion on templates/generic modules in #399.
